pkg/controllers/user/alert/watcher/event.go

```diff
@@ -91,3 +113,33 @@ func (l *EventWatcher) Sync(key string, obj *corev1.Event) (runtime.Object, erro
 
 	return nil, nil
 }
+
+type workloadFetcher struct {
+	unversionedClient rest.Interface
+}
+
+func (w *workloadFetcher) getOwnerRefferences(name, namespace, kind string) (rootName string, err error) {
+	obj := &unstructured.Unstructured{}
+	objKind := nameutil.GuessPluralName(kind)
+	objPrefix := resourcelink.ExportResourcePrefixMappings[strings.ToLower(objKind)]
+	if err = w.unversionedClient.Get().Prefix(objPrefix).Namespace(namespace).Resource(objKind).Name(name).Do().Into(obj); err != nil {
+		return "", errors.Wrapf(err, "get %s %s:%s failed", objKind, namespace, name)
+	}
+
+	allRef := obj.GetOwnerReferences()
+	if len(allRef) == 0 {
```

Comment:
There is a chance for workload which also has the owner reference information. Like the prometheus statefulSet from prometheus operator.